contracts/libraries/CurrencySettleTake.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.24;
+
+import {Currency, CurrencyLibrary} from "v4-core/types/Currency.sol";
```

Comment:
nit: Let's use consistent remappings. Otherwise this could leads to inconsistent build issues.
```suggestion
import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
```